enhancements/update/update-status-api.md

```diff
@@ -0,0 +1,398 @@
+
+#### Update Status Controller
+
+The Update Status Controller (USC) is a new component in OpenShift that will be deployed directly by the CVO and is being treated as its operand. This is uncommon in OpenShift, as the CVO typically deploys only second-level operators as its operands. In this model, the USC (providing cluster functionality) would normally be an operand, and we would need to invent a new cluster operator to manage it. Because the CVO is directly involved in updates, such an additional layer does not have value.
+
+The Update Status Controller will run a primary controller that will maintain the `UpdateStatus` resource. The resource has no `.spec`, so the controller will ensure the resource exists and its `.status` content is up-to-date and correct. It will determine the `status` subresource content by receiving and processing insights from the other controllers in the USC.
+
+The Update Status Controller will have two additional control loops, both serving as producers of insights for the given domain. One will monitor the control plane and will watch `ClusterVersion` and `ClusterOperator` resources. The other will monitor the node pools and will watch `MachineConfigPool` and `Node` resources. Both will report their observations as status or health insights to the primary controller so it can update the `UpdateStatus` resource. These control loops will reuse the existing cluster check code implemented in the client-side `oc adm upgrade status` prototype.
+
+To avoid inflating OpenShift payload images, the USC will be delivered in the same image as the CVO, so its code will live in the `openshift/cluster-version-operator` repository. The USC will be either a separate binary or a subcommand of the CVO binary (the CVO already has subcommands).
+
+> Note: As of Jan 7, the prototyping work showed the via-CVO delivery causes issues because the CVO image and manifests inside are treated specially by code that interacts with them (like hypershift [OCPBUGS-44438](https://issues.redhat.com/browse/OCPBUGS-44438) or release payload build [OCPBUGS-30080](https://issues.redhat.com/browse/OCPBUGS-30080)). Coupling with CVO still makes sense, but delivery through a standard additional payload image would prevent us from hitting subtle issues like the one mentioned above. What is the bar for the inclusion of a new image in the payload?
```

Comment:
For the bar to a new image, [Product builds and becoming part of an OpenShift release][1] requires an enhancement (which we have here :), and [the approval of image names for mirroring][2].  The latter requires [the OpenShift mirroring approvers][3].

[1]: https://docs.ci.openshift.org/docs/how-tos/onboarding-a-new-component/#product-builds-and-becoming-part-of-an-openshift-release
[2]: https://docs.ci.openshift.org/docs/how-tos/mirroring-to-quay/#mirroring-images
[3]: https://github.com/openshift/release/blob/master/core-services/image-mirroring/openshift/OWNERS

Reply:
Thanks for the links! I added the root link to the doc. There is another angle to the separate image / CVO: there's just a single release payload, no separate ones for TechPreview etc. Which means the image (only used by a small subset of clusters) would be necessary to e.g. mirror to disconnected environments. So I propose to continue shipping via CVO in TP and consider extracting to a dedicated image as a part of GA transition criteria.

Reply:
Thanks for the links! I added the root link to the doc. There is another angle to the separate image / CVO: there's just a single release payload, no separate ones for TechPreview etc. Which means the image (only used by a small subset of clusters) would be necessary to e.g. mirror to disconnected environments. So I propose to continue shipping via CVO in TP and consider extracting to a dedicated image as a part of GA transition criteria.

Addressed in https://github.com/openshift/enhancements/pull/1701/commits/353a4b5c5d9f48d7da62889dc6b055a74d54c84d